src/creator.rs

```diff
@@ -95,60 +98,241 @@ impl<H: Hasher> Creator<H> {
         }
     }
 
-    fn check_ready(&self, round: Round) -> bool {
-        if round == 0 {
-            return true;
-        }
-        if round > self.max_round {
-            warn!(target: "AlephBFT-creator", "{:?} Maximum round reached. Not creating another unit.", self.node_ix);
-            return false;
+    async fn ready_for_create(&mut self, round: Round) {
```

Comment:
I think the name `wait_until_ready` would suit better.

Reply:
Done